runtime/src/iree-amd-aie/driver/xrt/native_executable.h

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_AMD_AIE_DRIVER_XRT_NATIVE_EXECUTABLE_H_
+#define IREE_AMD_AIE_DRIVER_XRT_NATIVE_EXECUTABLE_H_
+
+#include <stdint.h>
+
+#include "iree/base/api.h"
+#include "iree/base/tracing.h"
+#include "iree/hal/api.h"
+#include "xrt/xrt_device.h"
+#include "xrt/xrt_kernel.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+// Object and launch parameters for a compute kernel.
+typedef struct iree_hal_xrt_kernel_params_t {
+  xrt::kernel* kernel;
```

Comment:
Can we put comments to explain what `kernel` and `instr`are? And how their lifetime is managed? (Given we are playing with C++ under the hood..)